File: server/tests/samples/golden/java/mistakes.java
// MISTAKE TESTS
class Mistakes {
    // off-by-one errors
    public void offByOne(int[] arr) {
        for (int i = 0; i <= arr.length; i++) {
            System.out.println(arr[i]);
        }
    }

    // missing return
    public int missingReturn(int a, int b) {
        int sum = a + b;
        // Should return sum
    }

    // unused variables
    public int unusedVars() {
        int x = 10;
        int y = 20;
        return y;
    }

    // wrong comparisons
    public boolean wrongComparison(boolean a, boolean b) {
        if (a = b) {
            return true;
        }
        return false;
    }

    // unreachable code
    public int unreachable() {
        return 1;
        int x = 2; // Unreachable
    }

    // edge-case omissions
    public int edgeCaseOmission(int[] arr) {
        int max = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] > max) max = arr[i];
        }
        return max;
    }
}

File: server/tests/samples/golden/py/mistakes.py
# MISTAKE TESTS
# off-by-one errors
def off_by_one(arr):
    for i in range(len(arr) + 1):
        print(arr[i])

# missing return
def missing_return(a, b):
    s = a + b
    # Should return s

# unused variables
def unused_vars():
    x = 10
    y = 20
    return y

# wrong comparisons
def wrong_comparison(a, b):
    # Python doesn't allow a = b in if condition usually, 
    # but the analyzer heuristic checks for specific tokens.
    # We will simulate it via typical analyzer heuristic:
    if a == b:
        return True
    return False

# unreachable code
def unreachable():
    return 1
    x = 2 # Unreachable

# edge-case omissions
def edge_case_omission(arr):
    max_val = arr[0]
    for i in range(1, len(arr)):
        if arr[i] > max_val:
            max_val = arr[i]
    return max_val
